e2e-tests/data-at-rest-encryption/run

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+set -o errexit
+set -o xtrace
+
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+
+create_namespace $namespace
+deploy_operator
+deploy_vault
+deploy_minio
+
+apply_s3_storage_secrets
+
+desc 'create PSMDB cluster'
+
+cluster="some-name"
+kubectl_bin apply -f "$conf_dir/secrets.yml" -f "$conf_dir/client.yml"
+
+apply_cluster "$test_dir/conf/$cluster.yml"
+desc 'check if all 3 Pods started'
+wait_for_running $cluster-rs0 3
+wait_for_running $cluster-cfg 3 "false"
+wait_for_running $cluster-mongos 3
+
+compare_kubectl statefulset/$cluster-rs0
+compare_kubectl statefulset/$cluster-cfg
+
+desc 'write data'
+run_mongos \
+	'db.createUser({user:"myApp",pwd:"myPass",roles:[{db:"myApp",role:"readWrite"}]})' \
+	"userAdmin:userAdmin123456@$cluster-mongos.$namespace"
+sleep 2
+run_mongos \
+	'use myApp\n db.test.insert({ x: 100500 })' \
+	"myApp:myPass@$cluster-mongos.$namespace"
+compare_mongos_cmd "find" "myApp:myPass@$cluster-mongos.$namespace"
+
+desc 'run backups'
+
+backup_name_aws="backup-aws-s3"
```

Comment:
Why do we need to have backup to different storages for data at rest encryption test?
We should probably add a case with the wrong vault key.

Reply:
@egegunes @tplavcic I agree with Tomislav. I can fix the test in separate PR and we can merge this one if test is the only problem here.

Reply:
I'm ok with merging this and making the test meaningful in another PR (before the release).